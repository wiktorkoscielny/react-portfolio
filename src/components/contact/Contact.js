import React from 'react'
import './contact.css'

// emailjs
import { useRef } from 'react';
import emailjs from 'emailjs-com'

// icons
import {MdOutlineMail} from 'react-icons/md'
import {RiMessengerLine} from 'react-icons/ri'
import {AiOutlineWhatsApp} from 'react-icons/ai'

const Contact = () => {

  const form = useRef();

  const sendEmail = (e) => {
    e.preventDefault();

    emailjs.sendForm(process.env.REACT_APP_SERVICEID_API_KEY, process.env.REACT_APP_TEMPLATEID_API_KEY, form.current, process.env.REACT_APP_PUBLICKEY_API_KEY)
      .then((result) => {
          alert('Email sent successfully!');
      }, (error) => {
          console.log(error.text);
          alert('Somethig went wrong, check the console');
      });

      e.target.reset();
  };

  return (
    <section id='contact'>
      <h5>Get in Touch</h5>
      <h2>Contact Me</h2>

      <div className='container contact__container'>

        {/* LEFT SIDE */}
        <div className='contact__options'>

          <article className='contact__option'>
            <MdOutlineMail className='contact__option-icon'/>
            <h4>Email</h4>
            <h5>loremipsum@gmail.com</h5>
            <a href='mailto:loremipsum@gmail.com' target='_blank'>Send Message</a>
          </article>
          <article className='contact__option'>
            <RiMessengerLine  className='contact__option-icon'/>
            <h4>Messenger</h4>
            <h5>loremipsum</h5>
            <a href='https://m.me/lorem.ipsum' target='_blank'>Send Message</a>
          </article>
          <article className='contact__option'>
            <AiOutlineWhatsApp className='contact__option-icon'/>
            <h4>WhatsApp</h4>
            <h5>+1234567</h5>
            <a href='https://api.whatsapp.com/send?phone=+4812345678'>Send Message</a>
          </article>

        </div>

        {/* RIGHT SIDE */}
        <form ref={form} onSubmit={sendEmail}>
          <input type='text' name='name' placeholder='Your Full Name' required />
          <input  type='email' name='email' placeholder='Your email' required/>
          <textarea name='message' rows='7' placeholder='Your Message' required></textarea>
          <button data-testid='button' type='submit' className='btn btn-primary'>Send Message</button>
        </form>

      </div>
    </section>
  )
}

export default Contact