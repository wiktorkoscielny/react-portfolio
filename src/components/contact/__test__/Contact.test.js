import React from "react";
import { unmountComponentAtNode } from "react-dom";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";

//component
import Contact from "./../Contact";

let container = null;
beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
});
afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});
it("renders without crashing", () => {
  render(<Contact />, container);
});

it("renders button correctly", () => {
  const { getByTestId } = render(<Contact />);
  expect(getByTestId("button")).toHaveTextContent("Send Message");
});

test("section has Contact Me text", () => {
  render(<Contact />);
  const textElement = screen.getByText(/Contact Me/i);
  expect(textElement).toBeInTheDocument();
});

//snapshot
it("matches snapshot", () => {
  const tree = renderer.create(<Contact />).toJSON();
  expect(tree).toMatchSnapshot();
});
