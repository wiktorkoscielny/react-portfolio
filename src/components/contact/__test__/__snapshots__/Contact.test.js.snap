// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<section
  id="contact"
>
  <h5>
    Get in Touch
  </h5>
  <h2>
    Contact Me
  </h2>
  <div
    className="container contact__container"
  >
    <div
      className="contact__options"
    >
      <article
        className="contact__option"
      >
        <svg
          className="contact__option-icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M22 6c0-1.1-.9-2-2-2H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6zm-2 0l-8 4.99L4 6h16zm0 12H4V8l8 5 8-5v10z"
          />
        </svg>
        <h4>
          Email
        </h4>
        <h5>
          loremipsum@gmail.com
        </h5>
        <a
          href="mailto:loremipsum@gmail.com"
          target="_blank"
        >
          Send Message
        </a>
      </article>
      <article
        className="contact__option"
      >
        <svg
          className="contact__option-icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M7.764 19.225c.59-.26 1.25-.309 1.868-.139.77.21 1.565.316 2.368.314 4.585 0 8-3.287 8-7.7S16.585 4 12 4s-8 3.287-8 7.7c0 2.27.896 4.272 2.466 5.676a2.8 2.8 0 0 1 .942 2.006l.356-.157zM12 2c5.634 0 10 4.127 10 9.7 0 5.573-4.366 9.7-10 9.7a10.894 10.894 0 0 1-2.895-.384.8.8 0 0 0-.534.039l-1.984.876a.8.8 0 0 1-1.123-.707l-.055-1.78a.797.797 0 0 0-.268-.57C3.195 17.135 2 14.617 2 11.7 2 6.127 6.367 2 12 2zM5.995 14.537l2.937-4.66a1.5 1.5 0 0 1 2.17-.4l2.336 1.75a.6.6 0 0 0 .723 0l3.155-2.396c.421-.319.971.185.689.633l-2.937 4.66a1.5 1.5 0 0 1-2.17.4l-2.336-1.75a.6.6 0 0 0-.723 0l-3.155 2.395c-.421.319-.971-.185-.689-.633z"
              fillRule="nonzero"
            />
          </g>
        </svg>
        <h4>
          Messenger
        </h4>
        <h5>
          loremipsum
        </h5>
        <a
          href="https://m.me/lorem.ipsum"
          target="_blank"
        >
          Send Message
        </a>
      </article>
      <article
        className="contact__option"
      >
        <svg
          className="contact__option-icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          t="1569683925316"
          version="1.1"
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <defs />
          <path
            d="M713.5 599.9c-10.9-5.6-65.2-32.2-75.3-35.8-10.1-3.8-17.5-5.6-24.8 5.6-7.4 11.1-28.4 35.8-35 43.3-6.4 7.4-12.9 8.3-23.8 2.8-64.8-32.4-107.3-57.8-150-131.1-11.3-19.5 11.3-18.1 32.4-60.2 3.6-7.4 1.8-13.7-1-19.3-2.8-5.6-24.8-59.8-34-81.9-8.9-21.5-18.1-18.5-24.8-18.9-6.4-0.4-13.7-0.4-21.1-0.4-7.4 0-19.3 2.8-29.4 13.7-10.1 11.1-38.6 37.8-38.6 92s39.5 106.7 44.9 114.1c5.6 7.4 77.7 118.6 188.4 166.5 70 30.2 97.4 32.8 132.4 27.6 21.3-3.2 65.2-26.6 74.3-52.5 9.1-25.8 9.1-47.9 6.4-52.5-2.7-4.9-10.1-7.7-21-13z"
          />
          <path
            d="M925.2 338.4c-22.6-53.7-55-101.9-96.3-143.3-41.3-41.3-89.5-73.8-143.3-96.3C630.6 75.7 572.2 64 512 64h-2c-60.6 0.3-119.3 12.3-174.5 35.9-53.3 22.8-101.1 55.2-142 96.5-40.9 41.3-73 89.3-95.2 142.8-23 55.4-34.6 114.3-34.3 174.9 0.3 69.4 16.9 138.3 48 199.9v152c0 25.4 20.6 46 46 46h152.1c61.6 31.1 130.5 47.7 199.9 48h2.1c59.9 0 118-11.6 172.7-34.3 53.5-22.3 101.6-54.3 142.8-95.2 41.3-40.9 73.8-88.7 96.5-142 23.6-55.2 35.6-113.9 35.9-174.5 0.3-60.9-11.5-120-34.8-175.6z m-151.1 438C704 845.8 611 884 512 884h-1.7c-60.3-0.3-120.2-15.3-173.1-43.5l-8.4-4.5H188V695.2l-4.5-8.4C155.3 633.9 140.3 574 140 513.7c-0.4-99.7 37.7-193.3 107.6-263.8 69.8-70.5 163.1-109.5 262.8-109.9h1.7c50 0 98.5 9.7 144.2 28.9 44.6 18.7 84.6 45.6 119 80 34.3 34.3 61.3 74.4 80 119 19.4 46.2 29.1 95.2 28.9 145.8-0.6 99.6-39.7 192.9-110.1 262.7z"
          />
        </svg>
        <h4>
          WhatsApp
        </h4>
        <h5>
          +1234567
        </h5>
        <a
          href="https://api.whatsapp.com/send?phone=+4812345678"
        >
          Send Message
        </a>
      </article>
    </div>
    <form
      onSubmit={[Function]}
    >
      <input
        name="name"
        placeholder="Your Full Name"
        required={true}
        type="text"
      />
      <input
        name="email"
        placeholder="Your email"
        required={true}
        type="email"
      />
      <textarea
        name="message"
        placeholder="Your Message"
        required={true}
        rows="7"
      />
      <button
        className="btn btn-primary"
        data-testid="button"
        type="submit"
      >
        Send Message
      </button>
    </form>
  </div>
</section>
`;
